### PR TITLE
chore: add error message when paymail is already taken

### PR DESCRIPTION
**Pull request by dangelovska**

### CODE REVIEW

File: brd-ios/breadwallet/Resources/InfoPlistLocalization/en.lproj/InfoPlist.strings [hunk 1]
```diff
@@ -3681,11 +3685,11 @@
 /* Limited assets message in add wallets popup */
 "Wallet.LimitedAssetsMessage" = "We currently only support the assets that are listed here. You cannot access other assets through this wallet at the moment.";
 
-"Wallet.one_day" = "24h";
+"Wallet.one_day" = "1D";
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/Resources/InfoPlistLocalization/en.lproj/InfoPlist.strings:3688]: why big letter? the old ones are all lower case.

File: brd-ios/breadwallet/Resources/InfoPlistLocalization/en.lproj/InfoPlist.strings [hunk 2]
```diff
@@ -3681,11 +3685,11 @@
...
 
 "Wallet.one_month" = "1m";
 
-"Wallet.one_week" = "7d";
+"Wallet.one_week" = "1W";
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/Resources/InfoPlistLocalization/en.lproj/InfoPlist.strings:3692]: why big letter? the old ones are all lower case.

File: brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressInteractor.swift [hunk 3]
```diff
@@ -41,7 +41,8 @@ class PaymailAddressInteractor: NSObject, Interactor, PaymailAddressViewActions
         presenter?.presentValidate(actionResponse: .init(email: viewAction.email,
                                                          isEmailValid: isEmailValid,
                                                          isEmailEmpty: isEmailEmpty,
-                                                         emailState: emailState))
+                                                         emailState: emailState,
+                                                         isPaymailTaken: false))
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressInteractor.swift:45]: this shouldn't be needed. can't you just know it if there is an error?

File: brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressModels.swift [hunk 4]
```diff
@@ -102,6 +102,7 @@ enum PaymailAddressModels {
             var isEmailValid: Bool
             var isEmailEmpty: Bool
             var emailState: DisplayState?
+            var isPaymailTaken: Bool
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressModels.swift:105]: why have a variable for each error?

File: brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressInteractor.swift (hunk 3 above)
Comment (dangelovska) [reply brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressInteractor.swift:45]: this is to distinguish which message to display, for paymail is different

File: brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressModels.swift (hunk 4 above)
Comment (dangelovska) [reply brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressModels.swift:105]: this is to distinguish which message to display, for paymail is different

File: brd-ios/breadwallet/Resources/InfoPlistLocalization/en.lproj/InfoPlist.strings (hunk 2 above)
Comment (dangelovska) [reply brd-ios/breadwallet/Resources/InfoPlistLocalization/en.lproj/InfoPlist.strings:3692]: I am not sure, this is not string updated by me.

File: brd-ios/breadwallet/Resources/InfoPlistLocalization/en.lproj/InfoPlist.strings (hunk 1 above)
Comment (dangelovska) [reply brd-ios/breadwallet/Resources/InfoPlistLocalization/en.lproj/InfoPlist.strings:3688]: I am not sure, this is not string updated by me.

File: brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressModels.swift (hunk 4 above)
Comment (kenanmamedoff) [reply brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressModels.swift:105]: can't this be distinguished by checking something else? like if self?.dataStore?.paymailAddress exists?

File: brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressInteractor.swift [hunk 5]
```diff
@@ -58,7 +59,19 @@ class PaymailAddressInteractor: NSObject, Interactor, PaymailAddressViewActions
...
-                self?.presenter?.presentError(actionResponse: .init(error: error))
+                guard (error as? NetworkingError)?.errorMessage == L10n.ErrorMessages.paymailTaken else {
+                    self?.presenter?.presentError(actionResponse: .init(error: error))
+                    return
+                }
+                
+                let isEmailValid = self?.dataStore?.paymailAddress?.isValidEmailAddress ?? false
+                let isEmailEmpty = self?.dataStore?.paymailAddress?.isEmpty == true
+                
+                self?.presenter?.presentValidate(actionResponse: .init(email: self?.dataStore?.paymailAddress,
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressInteractor.swift:70]: indentation is off

File: brd-ios/breadwallet/Resources/InfoPlistLocalization/en.lproj/InfoPlist.strings (hunk 1 above)
Comment (kenanmamedoff) [reply brd-ios/breadwallet/Resources/InfoPlistLocalization/en.lproj/InfoPlist.strings:3688]: this needs to be addressed

File: brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressModels.swift (hunk 4 above)
Comment (dangelovska) [reply brd-ios/breadwallet/src/Redesign/Scenes/PaymailAddress/PaymailAddressModels.swift:105]: I am adding the message in the presenter so I need to put it to the ActionResponse model

Screenshot 2023-06-22 at 15 16 37

**Review by kenanmamedoff — APPROVED** [review]

**Pull request merged**